app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -425,12 +425,28 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
     private fun setupNimbusObserver(nimbus: Observable<NimbusInterface.Observer>) {
         nimbus.register(object : NimbusInterface.Observer {
             override fun onUpdatesApplied(updated: List<EnrolledExperiment>) {
+                // To workaround a caching bug in Nimbus that appears when we try to query an
+                // experiment **before** `FxNimbus.initialize` is called we need to set the `api`
+                // value again so that we can still access the NimbusApi that is wrapped
+                // in `FxNimbus.initialize.getSdk`.
+                //
+                // We set this value here to minimize the race between setting the `api` value and
+                // the callers of FxNimbus.
+                // See: https://github.com/mozilla/application-services/issues/5075
+                FxNimbus.api = components.analytics.experiments
```

Comment:
@jhugman I don't think we can land either as a workaround, because it won't work [reliably](https://github.com/mozilla-mobile/fenix/pull/26228#discussion_r936194941). 

Moving this here has the advantage that the subsequent read for the tracking protection setting is on the same thread, but reads may still not be guaranteed. I am not confident landing this, but there are relatively easy solutions we should discuss. More below.

> The correct fix for this is still to upgrade the FML to v93.6.0 or later.
@jonalmeida can confirm, but I think he tried and it didn't solve the problem?

Reply:
> @jonalmeida can confirm, but I think he tried and it didn't solve the problem?

Yes, I tried the version from here with a local build: mozilla-mobile/android- components#12592

Verified I was using the new plugin by adding some print statements to show the new plugin version was showing.